compositor_render/src/scene/rescaler_component/layout.rs

```diff
@@ -0,0 +1,111 @@
+            }
+            (Some(child_width), Some(child_height)) => {
+                let scale = match self.mode {
+                    ResizeMode::Fit => {
+                        f32::min(size.width / child_width, size.height / child_height)
+                    }
+                    ResizeMode::Fill => {
+                        f32::max(size.width / child_width, size.height / child_height)
+                    }
+                };
+                self.layout_with_scale(size, child, pts, scale)
+            }
+        }
+    }
+
+    fn layout_with_scale(
+        &self,
+        size: Size,
+        child: &StatefulComponent,
+        pts: Duration,
+        scale: f32,
+    ) -> NestedLayout {
+        let (content, children, child_nodes_count) = match child {
+            StatefulComponent::Layout(layout_component) => {
+                let children_layouts = layout_component.layout(
+                    Size {
+                        width: size.width / scale,
+                        height: size.height / scale,
+                    },
+                    pts,
+                );
+                let child_nodes_count = children_layouts.child_nodes_count;
+                (
+                    LayoutContent::None,
+                    vec![children_layouts],
+                    child_nodes_count,
+                )
+            }
+            _non_layout => (StatefulLayoutComponent::layout_content(child, 0), vec![], 1),
+        };
+
+        let top = match self.vertical_align {
+            VerticalAlign::Top => 0.0,
+            VerticalAlign::Bottom => child
+                .height(pts)
+                .map(|height| size.height - (height * scale))
+                .unwrap_or(0.0),
+            VerticalAlign::Center | VerticalAlign::Justified => child
+                .height(pts)
+                .map(|height| (size.height - (height * scale)) / 2.0)
+                .unwrap_or(0.0),
+        };
+        let left = match self.horizontal_align {
+            HorizontalAlign::Left => 0.0,
+            HorizontalAlign::Right => child
+                .width(pts)
+                .map(|width| (size.width - (width * scale)))
+                .unwrap_or(0.0),
+            HorizontalAlign::Center | HorizontalAlign::Justified => child
+                .width(pts)
+                .map(|width| (size.width - (width * scale)) / (2.0))
+                .unwrap_or(0.0),
+        };
+        NestedLayout {
+            top,
+            left,
+            width: child
+                .width(pts)
+                .map(|child_width| child_width * scale)
+                .unwrap_or(size.width),
+            height: child
+                .height(pts)
+                .map(|child_height| child_height * scale)
+                .unwrap_or(size.height),
+            rotation_degrees: 0.0,
+            scale_x: scale,
+            scale_y: scale,
+            crop: None,
```

Comment:
not sure if rescaler should crop when using fill? currentlly it does not. any opinions?

Reply:
Option `visible` / `hidden`